From c9f2e25cc1ed3b9928212a920343374d306b6b2e Mon Sep 17 00:00:00 2001
From: Prasanth Ullattil <prasanth.ullattil@nokia.com>
Date: Fri, 18 Feb 2011 14:49:11 +0100
Subject: [PATCH 6/7] Fix link issues on Mac

---
 Source/WebKit/qt/QtWebKit.pro |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Source/WebKit/qt/QtWebKit.pro b/Source/WebKit/qt/QtWebKit.pro
index 7f6943d..6b1354f 100644
--- a/Source/WebKit/qt/QtWebKit.pro
+++ b/Source/WebKit/qt/QtWebKit.pro
@@ -451,7 +451,7 @@ contains(DEFINES, ENABLE_GEOLOCATION=1) {
             QMAKE_BUNDLE_DATA += FRAMEWORK_HEADERS
         }
 
-        QMAKE_LFLAGS_SONAME = "$${QMAKE_LFLAGS_SONAME}$${DESTDIR}$${QMAKE_DIR_SEP}"
+        !contains(QT_CONFIG, modular):QMAKE_LFLAGS_SONAME = "$${QMAKE_LFLAGS_SONAME}$${DESTDIR}$${QMAKE_DIR_SEP}"
     }
 }
 
-- 
1.7.2.2

